pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -231,7 +231,9 @@ func TestSyncDeploymentDontDoAnythingDuringDeletion(t *testing.T) {
 	now := metav1.Now()
 	d.DeletionTimestamp = &now
 	f.dLister = append(f.dLister, d)
+	f.objects = append(f.objects, d)
 
+	f.expectUpdateDeploymentStatusAction(d)
```

Comment:
We update the status of the Deployment and there was a mutation in sync.go#L585 that was blocking this update (we were mutating both the old and the new version of the status hence there was no difference and nothing was updated).